aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -109,6 +109,24 @@ def handle_unrecoverable_failure(self, node):
             self.report_error_handled(node, 'unrecoverable error, aborting...')
             return ProcessHandlerReport(True, self.exit_codes.ERROR_UNRECOVERABLE_FAILURE)
 
+    @process_handler(priority=610, exit_codes=PhCalculation.exit_codes.ERROR_SCHEDULER_OUT_OF_WALLTIME)
+    def handle_scheduler_out_of_walltime(self, node):
+        """Handle `ERROR_SCHEDULER_OUT_OF_WALLTIME` exit code: decrease the max_secondes and restart from scratch."""
+
+        # decrease max_seconds a lot in order to make sure the calculation shut down neatly before reaching the
+        # scheduler wall time
+        factor = 0.5
+        max_seconds = self.ctx.inputs.parameters.get('INPUTPH', {}).get('max_seconds', None)
+        max_seconds_new = max_seconds * factor
+
+        self.ctx.restart_calc = node
+        self.ctx.inputs.parameters.setdefault('INPUTPH', {})['recover'] = False
```

Comment:
I am not too familiar with `ph.x` so I am not sure if this is the correct way to do it, but it could very well be. Just saying that I cannot really sign off on this as I have no idea. I would advise that we ask someone that is knowledgeable on `ph.x` to validate this strategy.